appinventor/components/src/com/google/appinventor/components/runtime/ActivityStarter.java

```diff
@@ -359,6 +362,30 @@ public String ResultUri() {
     return "";
   }
 
+  /**
+   * Specifies the list of key-value pairs that will be passed as extra data to the activity.
+   */
+  @SimpleProperty
+  public void Extras(YailList pairs) {
```

Comment:
just did. i thought it was unnecessary. but users may get the value before
setting it. the old code will return null to them

On Tue, Jun 23, 2015 at 10:04 PM, hal <notifications@github.com> wrote:

> In
> appinventor/components/src/com/google/appinventor/components/runtime/ActivityStarter.java
> <https://github.com/mit-cml/appinventor-sources/pull/424#discussion_r33043738>
> :
>
> > @@ -359,6 +362,30 @@ public String ResultUri() {
> >      return "";
> >    }
> >
> > +  /**
> > +   * Specifies the list of key-value pairs that will be passed as extra data to the activity.
> > +   */
> > +  @SimpleProperty
> > +  public void Extras(YailList pairs) {
>
> I don't think you've initialized Extras. You should set it to an empty
> (Yail) list when the Activity Starter component is constructed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/424/files#r33043738>.
>